app/main.py

```diff
@@ -1,6 +1,16 @@
-from typing import Callable
+from typing import Callable, Any
+from functools import wraps
 
 
 def cache(func: Callable) -> Callable:
-    # Write your code here
-    pass
+    result = {}
```

Comment:
The `result` dictionary is defined at the function level, which means it will be shared across all instances of the decorated function. This is fine if the decorator is used on a single function, but if used on multiple functions, their results will be mixed. Consider using a more robust solution if the decorator is intended for multiple functions.